[feconi1024] iP

feconi1024 opened the pull request:
# Fairy

> Hello, *Master*. 

**Fairy** helps you to organize your tasks in a good manner. It's
- command-based
- suitable for keyboard-lovers
- fast to install

How to use such convenient tool? In 3 steps:
1. Download it from [here](https://github.com/feconi1024/ip/releases).
2. Double-click it.
3. You're all set! 😉 

Features:

- [x] managing todos, deadlines and events
- [x] search tasks
- [ ] edit tasks (coming soon)

My favorite part of code in the project is the `CommandParser`:

```java
if (cmdSplit.length > 1) {
    String rest = cmdSplit[1];
    // Split the rest on any occurrence of whitespace followed by /word
    String[] parts = rest.split(REGEX_COMMAND_FLAG_PARSE);
    for (String part : parts) {
        String trimmedPart = part.trim();
        if (!trimmedPart.isEmpty()) {
            result.add(trimmedPart);
        }
    }
}
```

Actually it was written by AI 😆 .

j-lum committed:
Merge pull request #3 from j-lum/inline-javafx-tutorials

Inline JavaFX tutorials

j-lum committed:
Update Gradle tutorial to reflect new checkstyle config

j-lum committed:
Fix typo referring to `HelloWorld`

j-lum committed:
Add images to JavaFX tutorial 3

j-lum committed:
Merge pull request #7 from j-lum/tutorial-fix

Address issues #5 and #6

j-lum committed:
Remove references to DukeStub

j-lum committed:
Merge pull request #8 from j-lum/tutorial-fix

Remove references to DukeStub

j-lum committed:
Change indentation in FXML samples to match AB-3

j-lum committed:
Merge pull request #14 from j-lum/fxml-indent

Change indentation in FXML samples to match AB-3

j-lum committed:
JavaFX tutorial: Support cross-platform JARs

The OpenJFX plugin expects applications to be modular and bundled
with jlink, resulting in fat jars that are not cross-platform. Let's
manually include the required dependencies so that shadow can package
them properly.

j-lum committed:
Merge pull request #35 from j-lum/xplatformjfx

JavaFX tutorial: Support cross-platform JARs

jiachen247 committed:
Remove unnecessary override annotation

Remove extra '@Override' at the start of the code example.

j-lum committed:
Merge pull request #41 from jiachen247/patch-2

Remove unnecessary override annotation

Commit:
Remove generic tutorials

Let's remove generic tutorials from this repo as they are not specific
to this project tempalte

Commit:
Add text-ui-test files

Commit:
Add Gradle support

damithc committed:
README.md: Fix typo of => or

Commit:
Improve runtest.* scripts

* Improve the javac command to work for multiple java files.
* In runtest.bat, delete ACTUAL.TXT only if it exists.

Commit:
README: Update setup instructions to match the latest IDE version

Commit:
README.md: Tweak template text

* Let's replace 'Feature 1' with 'Feature-ABC'
  reason: some students kept the 'Feature 1' in the final UG.
* Let's tweak the template for features and usage
  reasons: to fix grammar, to distinguish between output and outcome

damithc committed:
README: Mention the project language level setting

seanleong339 committed:
.gitignore: Fix ACTUAL.txt -> ACTUAL.TXT  (#73)

text-ui-test/runtest.bat|sh scripts generate a file ACTUAL.TXT.
However, .gitignore uses ACTUAL.txt, which means the generated
file will not be ignored by Git on non-Windows OS.

Let's update .gitignore as ACTUAL.txt -> ACTUAL.TXT

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
README: Update the JDK version 11 -> 17

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

damithc committed:
README: Ask to keep source folder structure

damithc committed:
README: Use correct item number in list

feconi1024 committed:
Replace Duke.java by Fairy.java. Add skeleton structure. Implement greet and exit

feconi1024 committed:
Customize output format and message

feconi1024 committed:
Added echo functionality

feconi1024 committed:
Modify variable naming to align with standard

feconi1024 committed:
Add store and display of tasks

feconi1024 committed:
Add Task class and mark as done functionality

feconi1024 committed:
Add Todo class

feconi1024 committed:
Add Deadline class

feconi1024 committed:
Add Event class

feconi1024 committed:
Add command parser and 3 more task types

feconi1024 committed:
Adapt Automated Text UI Testing

feconi1024 committed:
Add exception handling

feconi1024 committed:
Add delete feature

jniaorx commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+public class Deadline extends Task {
+    private final String endTime;
+
+    public Deadline(String taskName, String endTime){
```
Perhaps there should be a space between the closing bracket and the opening curly braces?

jniaorx commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+public class Fairy {
+    private static final String NAME = "Fairy";
+    private static final Scanner SC = new Scanner(System.in);
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
```
Good job on naming these as constants and using uppercase for them!

jniaorx commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+                        argumentExceptionMessage();
+                    }
+                    break;
+                case "delete":
```
Good job on using switch statement for this, but maybe the case clauses do not need indentation?

Frozennfishh replied:
> Good job on using switch statement for this, but maybe the case clauses do not need indentation?

Yep just take note that case blocks should be the same indent as "case", you can try to change the indentation setting in your IDE for easier editing in the future!

jniaorx commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+                        }
+                    }
+                    break;
+                default:
```
Well done, but perhaps the default clause shouldn't need an indentation just like case clauses?

jniaorx commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+        done = false;
+    }
+
+    public String getTaskName() {
```
Great job on using get for the naming!

jniaorx commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+        return taskName;
+    }
+
+    public void setDo() {
```
Maybe it would be better if this method is named like boolean by using a prefix such as is, has, was?

Frozennfishh replied:
> Maybe it would be better if this method is named like boolean by using a prefix such as is, has, was?

For setters i think should be fine, but since your variable is "done", maybe use ```setDone()``` instead?

jniaorx commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+            return;
+        }
+        String output = "Master, here are the tasks in your list:\n";
+        for (int i = 0; i < TASKS.size(); i++) {
```
Great job for following the suggested use of whitespace in Java code as well as the naming of variables in a for loop!

jniaorx reviewed (commented):
Overall, I find that your code was pretty well-written and was easy to read except for some minor indentation and naming mistakes. Good job and keep up the good work!

feconi1024 committed:
Add index out of bounds exception message

Frozennfishh commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+        }
+
+        return result;
+    }
```
Perhaps a parser class can be used for this? For easier class management and readability

Frozennfishh commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+
+    private static void argumentExceptionMessage() {
+        printStandardFormat("Argument exception: No enough arguments.");
+    }
```
Prints could also be done in a UI class as stated in the project website, maybe can try implementing in the future? 😄

Frozennfishh commented on src/main/java/Fairy.java:
```diff
@@ -0,0 +1,216 @@
+        Task removedTask = TASKS.remove(index - 1);
+        printStandardFormat("Yes, Master. I've removed this task from your list:\n" +
+                removedTask.toString().indent(2) + "\nThere are " + TASKS.size() + " tasks in your list now.");
+    }
```
For tasks methods, maybe could be derived from a parent task class and extend from there for the different type of classes? Better use of OOP and easier for adding more specialized commands for new tasks in the future!

Frozennfishh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+            return "[X] " + taskName;
+        } else {
+            return "[ ] " + taskName;
+        }
```
Could potentially use the "done" boolean to print out the string accordingly, i.e.
```return "[" + done ? "X" : " " + "] " + taskName;```

Frozennfishh reviewed (commented):
LGTM for up to Week 2/3 requirements, but still have much to work on! 😉 
Some nits for coding standards, but mostly still waiting for week 3/4 requirements (OOP and packages), but good job so far! 👍

feconi1024 committed:
Add file saving feature

feconi1024 committed:
Modify gitignore to include saved data

feconi1024 committed:
Add file read feature

feconi1024 committed:
Merge branch 'branch-Level-7'

feconi1024 committed:
Eliminate message print when add from file

feconi1024 committed:
Add file lines load count

feconi1024 committed:
Merge branch 'feature/optimize-save'

feconi1024 committed:
Add minimum viable DateTime functionality

feconi1024 committed:
Change DateTime exception handling method name

feconi1024 committed:
Add exception handling for illegal times

feconi1024 committed:
Add search task by date

feconi1024 committed:
Merge branch 'branch-Level-8'

feconi1024 committed:
Separate Ui functions as new class from Fairy

feconi1024 committed:
Separate Storage functions as new class from Fairy

feconi1024 committed:
Separate CommandParser as new class from Fairy

feconi1024 committed:
Separate TaskList as new class from Fairy

feconi1024 committed:
Modify string concatenation method

feconi1024 committed:
Delete useless import statements

feconi1024 committed:
Add Command as abstract class

feconi1024 committed:
Add mark commands as classes

feconi1024 committed:
Separate ListCommand with new class from Fairy with new exception and utility

feconi1024 committed:
Modify Command execute signature for direct print

feconi1024 committed:
Separate TodoCommand as new class from Fairy

feconi1024 committed:
Separate add commands as new classes from Fairy

feconi1024 committed:
Separate delete command as new class from Fairy

feconi1024 committed:
Separate search by date command as new class from Fairy

feconi1024 committed:
Modify Ui from static to dynamic

feconi1024 committed:
Modify TaskList to align with design

feconi1024 committed:
Modify Storage to align with design

feconi1024 committed:
Modify command classes to align with design

feconi1024 committed:
Modify command parser to align with design

feconi1024 committed:
Separate exit command as new class from Fairy

feconi1024 committed:
Modify main class to align with design

feconi1024 committed:
Finalize MoreOOP design

feconi1024 committed:
Merge branch 'branch-A-MoreOOP'

feconi1024 committed:
Transfer messages in Ui to separate class

feconi1024 committed:
Substitute with static strings in Ui

feconi1024 committed:
Substitute with static strings in add commands

feconi1024 committed:
Substitute with static strings in list commands

feconi1024 committed:
Rectify error output in search by date

feconi1024 committed:
Merge branch 'feature/Enhance-OOP'

feconi1024 committed:
Divide classes into packages and rectify command classes

feconi1024 committed:
Rectify common classes to align with package design

feconi1024 committed:
Rectify parser classes to align with package design

feconi1024 committed:
Rectify storage classes to align with package design

feconi1024 committed:
Rectify task classes to align with package design

feconi1024 committed:
Rectify ui classes to align with package design

feconi1024 committed:
Rectify error in displaying events

feconi1024 committed:
Merge branch 'branch-A-Packages'

feconi1024 committed:
Merge branch 'branch-add-gradle-support'

feconi1024 committed:
Rectify chatbot name

feconi1024 committed:
Rectify chatbot name in Ui

feconi1024 committed:
Update build.gradle configuration for Fairy

feconi1024 committed:
Modify gradlew.bat to prevent error codes

feconi1024 committed:
Add JUnit test

feconi1024 committed:
Merge branch 'branch-A-JUnit'

feconi1024 committed:
Update gradle file to include version info in Jar output

feconi1024 committed:
Update .gitignore to prevent updating JAR file

feconi1024 committed:
Rectify JAR packaging errors in the previous commit

feconi1024 committed:
Merge branch 'branch-A-Jar'

feconi1024 committed:
Complete JavaDoc for Fairy

feconi1024 committed:
Complete JavaDoc for command package

feconi1024 committed:
Complete JavaDoc for common package

feconi1024 committed:
Complete JavaDoc for exception package

feconi1024 committed:
Complete JavaDoc for parser package

feconi1024 committed:
Complete JavaDoc for storage package

feconi1024 committed:
Rectify error in messages and variable names

feconi1024 committed:
Rectify command package for JavaDoc code display

feconi1024 committed:
Rectify utils package for JavaDoc code display

feconi1024 committed:
Complete JavaDoc for task package

feconi1024 committed:
Complete JavaDoc for UI package

feconi1024 committed:
Rectify code for naming standard

feconi1024 committed:
Rectify code for layout standard

feconi1024 committed:
Rectify code for statement standard

feconi1024 committed:
Add find by keyword functionality

feconi1024 committed:
Rectify class variables with public scope to private

feconi1024 committed:
Rectify all overridden methods that have no @Override

feconi1024 committed:
Merge branch 'branch-A-CodingStandard'

feconi1024 committed:
Resolve merge conflict between master and A-JavaDoc

feconi1024 committed:
Resolve merge conflict between master and Level-9

feconi1024 committed:
Final check for week 3

feconi1024 committed:
Add checkstyle support

feconi1024 committed:
Merge branch 'branch-A-CheckStyle'

feconi1024 committed:
Add checkstyle

feconi1024 committed:
Transfer GUI classes and files

feconi1024 committed:
Complete command package rectification

feconi1024 committed:
Complete basic integration of GUI

feconi1024 committed:
Add show of greeting and exit messages and exit functionality

feconi1024 committed:
Check coding standard

feconi1024 committed:
Merge branch 'branch-Level-10'

feconi1024 committed:
Update build.gradle

feconi1024 committed:
Add assertions to code

Assertion is important to ensure the invariants in code.

Let's add assertions to TaskList and Storage classes.

These classes contains invariants regarding filepath and legal values.

Therefore use assertions there is appropriate.

feconi1024 committed:
Reduce magic numbers

feconi1024 committed:
Eliminate unused codes

feconi1024 committed:
Simplify complicated expressions

feconi1024 committed:
Final check for code quality

feconi1024 committed:
Eliminate unused import statements

feconi1024 committed:
Merge pull request #2 from feconi1024/branch-A-Assertions

Add assertion statements

feconi1024 committed:
Merge branch 'master' into branch-A-CodeQuality

feconi1024 committed:
Merge pull request #3 from feconi1024/branch-A-CodeQuality

Improve code quality

feconi1024 committed:
Add task type with fixed duration

Many tasks comes with no specific time point, but with confirmed time location.

Adding this kind of task help users more.

Let's add new FixedDurationTask class with duration field.

feconi1024 committed:
Add process routine in task list class

feconi1024 committed:
Add fixdur command

feconi1024 committed:
Add exception handling

feconi1024 committed:
Merge branch 'branch-B-FixedDurationTasks'

feconi1024 committed:
Check for error and plagiarisms

feconi1024 committed:
Add reference and change avatar of chatbot

feconi1024 committed:
Merge branch 'branch-Final-Check'

feconi1024 committed:
Change avatar

feconi1024 committed:
Add automatic element resizing

feconi1024 committed:
Add css support

feconi1024 committed:
Add dialog box flip effect

feconi1024 committed:
Add references for css files

feconi1024 committed:
Change user portrait

feconi1024 committed:
Merge branch 'branch-A-BetterGui'

feconi1024 committed:
Rectify the program to save file when Gui window is closed

feconi1024 committed:
Merge branch 'fixbug/GuiExitSave'

feconi1024 committed:
Update UI sample picture

feconi1024 committed:
Complete user guide

feconi1024 committed:
Merge branch 'branch-A-UserGuide'

feconi1024 committed:
Change main README.md

feconi1024 committed:
Update README.md
